### PR TITLE
[DROOLS-2527] Add support for Unit in simulation fluent API

### PR DESCRIPTION
**Pull request by danielezonca**

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/command/AddDataSourceCommand.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private final List<String> names;
+    private final DataSource<E> values;
+
+    public AddDataSourceCommand(Class<E> clazz, List<String> names, DataSource<E> values) {
+        this.clazz = clazz;
+        this.names = names;
+        this.values = values;
+    }
+
+    public Void execute(Context context) {
```
Comment (baldimir) [inline drools-core/src/main/java/org/drools/core/command/AddDataSourceCommand.java]: Add Override annotation please.

File: drools-core/src/main/java/org/drools/core/command/BindVariableToUnitCommand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public BindVariableToUnitCommand(String variableName, Function<Context, ?> lazyValue) {
+        this.lazyValueProvider = lazyValue;
+        this.variableName = variableName;
+    }
+
+    public <E> BindVariableToUnitCommand(String name, E variable) {
+        this(name, context -> variable);
+    }
+
+    public Void execute(Context context) {
```
Comment (baldimir) [inline drools-core/src/main/java/org/drools/core/command/BindVariableToUnitCommand.java]: Add Override annotation please.

File: drools-core/src/main/java/org/drools/core/command/NewRuleUnitExecutorCommand.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    public NewRuleUnitExecutorCommand(String sessionId) {
+        this.sessionId = sessionId;
+    }
+
+    public NewRuleUnitExecutorCommand(ReleaseId releaseId, String sessionId) {
+        this.sessionId = sessionId;
+        this.releaseId = releaseId;
+    }
+
+    public RuleUnitExecutor execute(Context context) {
```
Comment (baldimir) [inline drools-core/src/main/java/org/drools/core/command/NewRuleUnitExecutorCommand.java]: Add Override annotation please.

File: drools-core/src/main/java/org/drools/core/command/RunUnitCommand.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    public RunUnitCommand(Class<T> unitClass) {
+        this.unitClass = unitClass;
+    }
+
+    public RunUnitCommand(Supplier<T> unitSupplier) {
+        this.unitSupplier = unitSupplier;
+    }
+
+    public Integer execute(Context context) {
```
Comment (baldimir) [inline drools-core/src/main/java/org/drools/core/command/RunUnitCommand.java]: Add Override annotation please.

**Comment by baldimir** [comment]
ok to test

**Review by baldimir — APPROVED** [review]

**Review by jomarko — APPROVED** [review]

**Pull request merged**